src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,118 @@
+     *
+     * @param taskList The taskList to be saved.
+     */
+    public void Save(TaskList taskList) {
+        try {
+            ArrayList<Task> arr = taskList.getEntire();
+            PrintWriter printWriter = new PrintWriter(FileName);
+            StringBuilder sb = new StringBuilder();
+            for (Task task : arr) {
+                sb.append(task.getLogo());
+                sb.append(",");
+                sb.append(task.checkDone() ? "1" : "0");
+                sb.append(",");
+                sb.append(task.getDescription());
+                if (task instanceof Event) {
+                    sb.append(",");
+                    sb.append(((Event) task).getAt());
+                }
+                if (task instanceof Deadline) {
+                    sb.append(",");
+                    sb.append(((Deadline) task).getBy());
+                }
+                sb.append(System.lineSeparator());
+            }
+            printWriter.write(sb.toString());
+            printWriter.close();
+        } catch (FileNotFoundException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Loads a previously stored taskList from the hard disk.
+     * The filename of the file is stored in the attribute Filename of the class.
+     * If the file does not exist, an empty taskList is instantiated and returned.
+     *
+     * @return The previously stored taskList, or if it is unavailable, an empty taskList.
+     */
+    public static ArrayList<Task> Load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File file = new File(FileName);
+        try {
+            Scanner sc = new Scanner(file);
+            System.out.println("Past tasks found. Use command \"list\" to list previous tasks.");
+            sc.useDelimiter(",");
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                List<String> splitLine = Arrays.stream(line.split(","))
+                        .map(String::trim)
+                        .collect(Collectors.toList());
+
+                String taskType = splitLine.get(0);
+                boolean isDone = splitLine.get(1).equals("1");
+                String description = splitLine.get(2);
+
+                Task toAdd;
+                switch (taskType) {
+                    case "T":
+                        toAdd = new Todo(description);
+                        break;
+                    case "D":
+                        LocalDateTime by = LocalDateTime.parse(splitLine.get(3));
+                        toAdd = new Deadline(description, by);
+                        break;
+                    case "E":
+                        LocalDateTime at = LocalDateTime.parse(splitLine.get(3));
+                        toAdd = new Event(description, at);
+                        break;
+                    default:
+                        toAdd = new Task(description);
+                        break;
+                }
```

Comment:
Do follow the module coding style for switch block indentations!

https://se-education.org/guides/conventions/java/intermediate.html#layout